lib/logstash/filters/drop.rb

```diff
@@ -18,7 +18,20 @@
 # `debug`. This will cause all events matching to be dropped.
 class LogStash::Filters::Drop < LogStash::Filters::Base
   config_name "drop"
-
+  # Drop all the events within a pre-configured percentage.
+  #
+  # This is useful if you just need a sample but not the whole.
+  #
+  # Example, to only drop around 40% of the events that have the field loglevel wiht value "debug".
+  #
+  #     filter {
+  #       if [loglevel] == "debug" { 
+  #         drop { 
+  #           sample => 40
+  #         } 
+  #       }
+  #     }
+  config :sample, :validate => :number, :default => 100
```

Comment:
Can we call this `percentage` ? "sample => 50" sounds weird in my head when I read it.

Thoughts on an alternative name?